pkg/jobrunaggregator/jobrunaggregatorapi/types_row_historical_data.go

```diff
@@ -0,0 +1,95 @@
+package jobrunaggregatorapi
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type HistoricalDataRow struct {
+	Name         string
+	Release      string
+	FromRelease  string
+	Platform     string
+	Architecture string
+	Network      string
+	Topology     string
+	P95          string
+	P99          string
+	Type         string
+}
+
+func (c *HistoricalDataRow) UnmarshalJSON(data []byte) error {
+	var values struct {
+		AlertName    string
+		BackendName  string
+		Release      string
+		FromRelease  string
+		Platform     string
+		Architecture string
+		Network      string
+		Topology     string
+		P95          string
+		P99          string
+	}
+	err := json.Unmarshal(data, &values)
+	if err != nil {
+		return err
+	}
+	c.Name = values.BackendName
+	c.Type = "disruption"
+	if values.AlertName != "" {
+		c.Name = values.AlertName
+		c.Type = "alert"
+	}
```

Comment:
I think it might be good to go with two separate types here, one for disrupt and one for alerts. These marshal functions would no longer be needed and the APIs are a little cleaner for anyone reading or using them, no confusion about what fields will or won't be set. More flexibility to adjust them individually in future as well.

If there is any major downside to that though let me know.

Reply:
Removed that entirely and separated them out into two different structs